From 08cf7bd0af8a44938bbcbc5027ed7b1031442ae6 Mon Sep 17 00:00:00 2001
From: Jonathan Hsieh <jmhsieh@apache.org>
Date: Wed, 24 Apr 2013 18:08:28 +0000
Subject: [PATCH 82/96] HBASE-8381 TestTableInputFormatScan on Hadoop 2 fails because YARN kills our applications

This patch bumps up the vmem-pmem ratio so that yarn's resource checker does not kill jobs that exceed the vmem (virtual memory!) allocation limit.  Related to HBASE-6330.

git-svn-id: https://svn.apache.org/repos/asf/hbase/branches/0.94@1471570 13f79535-47bb-0310-9956-ffa450edef68

Reason: Unit Test Failures
Author: Jonathan Hsieh
Ref: CDH-11675
---
 .../apache/hadoop/hbase/HBaseTestingUtility.java   |    6 +++++-
 1 files changed, 5 insertions(+), 1 deletions(-)

diff --git a/src/test/java/org/apache/hadoop/hbase/HBaseTestingUtility.java b/src/test/java/org/apache/hadoop/hbase/HBaseTestingUtility.java
index d2fb218..08eb554 100644
--- a/src/test/java/org/apache/hadoop/hbase/HBaseTestingUtility.java
+++ b/src/test/java/org/apache/hadoop/hbase/HBaseTestingUtility.java
@@ -1417,6 +1417,11 @@ public class HBaseTestingUtility {
     }
     System.setProperty("hadoop.log.dir", logDir);
     c.set("mapred.output.dir", tmpDir);
+
+    // Tests were failing because this process used 6GB of virtual memory and was getting killed.
+    // we up the VM usable so that processes don't get killed.
+    conf.setFloat("yarn.nodemanager.vmem-pmem-ratio", 8.0f);
+
     mrCluster = new MiniMRCluster(servers,
       FileSystem.get(conf).getUri().toString(), 1);
     LOG.info("Mini mapreduce cluster started");
@@ -1743,7 +1748,6 @@ public class HBaseTestingUtility {
     return startedServer;
   }
 
-
   /**
    * Make sure that at least the specified number of region servers
    * are running. We don't count the ones that are currently stopping or are
-- 
1.7.0.4

